### DISCUSSION TITLE
Macbook air with single trackpad button: how to right-click?

### DISCUSSION THREAD

**Discussion by reefsofts-rhosds** [elementary/triage#662] [Q&A]
# Macbook air with single trackpad click: how to right-click?
Hello,
I own a  MacBookAir7,2 that has only left click on its trackpad (only one button that is left click-mapped). I would like to have Ctrl+LeftClick mapped to right click, but i can't figure out how. I run elementaryOS 8. Could you please help me out?

**Comment by jeremypw** [comment]
Control-clicking is used by some apps for other purposes so I am not sure it is desirable, or possible, to do it your way through the user interface. However there are other builtin ways.  You can go into mouse-touchpad settings and under "Clicking" turn on "Long press to secondary click".  Configure the desired press time.  After doing this, if you hold down the primary button longer than the configured time and then release a secondary click is produced.

Another way (and the one I use) is to go into "Touchpad" settings, turn on "Tap to Click" and under "Physical Secondary Clicking" select the desired method (I use "Multitouch").  Then tapping quickly on the touchpad with two fingers produces a secondary click (one finger produces a primary click and three produces a middle click).  The other method "Areas" doesn't seem to work for me but that may be hardware related.

**Reply by reefsofts-rhosds** [reply]
Hello Jeremy,
None of them work for me sadly (the multitouch way didn't show the context menu in Files, same for the long-clicking way). I am not scared using the terminal because i used Arch before (btw). I said Ctrl+LC but Fn+LC, Shift+LC or Atl+LC or even Cmd+LC would work for me as well.

**Reply by jeremypw** [reply]
That's a pity - I guess the hardware or driver does not support those features.   I am surprised that the long-press method at least does not work.  When you hold down the button you should get a circular animation at the cursor, the secondary click will occur if you release the button *after* the animation completes.   Maybe try re-login or re-boot??

**Reply by jeremypw** [reply]
Are you running elementary on bare metal or in a virtual machine?

**Reply by jeremypw** [reply]
I can confirm it works on my hardware with OS8 in a virtual machine on a generic laptop.

**Comment by jeremypw** [comment]
For opening a context menu (the main use for secondary click) you can use the "Menu" key if you have one or `<Shift>F10` may work for you (it does for me).

**Comment by reefsofts-rhosds** [comment]
i run it on bare metal, the keyboard shortcuts don't work for some reason.



-- 
 Youri Ledoux Kharlamov
LFA de Strasbourg, Club informatux
Me contacter par chat
Contacter le club informatux (uniquement les jeudis de 12h30 à 13h30):
https://1698.3cx.cloud/youlksoft


Am Sonntag, Dezember 22, 2024 19:52 CET, schrieb Jeremy Wootten ***@***.***>:

 
 
For opening a context menu (the main use for secondary click) you can use the "Menu" key if you have one or <Shift>F10 may work.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/orgs/elementary/discussions/662#discussioncomment-11644248", "url": "https://github.com/orgs/elementary/discussions/662#discussioncomment-11644248", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by jeremypw** [reply]
You could try raising it on the elementary community Discord server.  Someone with the same hardware maybe able to help you better.

**Comment by ryonakano** [comment]
I have a 11-inch MacBook Air 2010 and when I ran elementary OS on it previously IIRC I was able to perform a "right-click" by tapping the trackpad with two fingers, which is the same behavior as macOS.
